contracts/lib/time_series/sorting.cairo

```diff
@@ -0,0 +1,154 @@
+        return _merge(
+            left_arr_len,
+            left_arr,
+            right_arr_len,
+            right_arr,
+            sorted_arr,
+            current_ix + 1,
+            left_arr_ix,
+            right_arr_ix + 1,
+        )
+    end
+
+    if right_arr_len == right_arr_ix:
+        assert sorted_arr[current_ix] = left_arr[left_arr_ix]
+        return _merge(
+            left_arr_len,
+            left_arr,
+            right_arr_len,
+            right_arr,
+            sorted_arr,
+            current_ix + 1,
+            left_arr_ix + 1,
+            right_arr_ix,
+        )
+    end
+
+    let left_val = left_arr[left_arr_ix].tick
+    let right_val = right_arr[right_arr_ix].tick
+    let (is_left) = is_le(left_val, right_val)
+
+    if is_left == 1:
+        assert sorted_arr[current_ix] = left_arr[left_arr_ix]
+        return _merge(
+            left_arr_len,
+            left_arr,
+            right_arr_len,
+            right_arr,
+            sorted_arr,
+            current_ix + 1,
+            left_arr_ix + 1,
+            right_arr_ix,
+        )
+    else:
+        assert sorted_arr[current_ix] = right_arr[right_arr_ix]
+        return _merge(
+            left_arr_len,
+            left_arr,
+            right_arr_len,
+            right_arr,
+            sorted_arr,
+            current_ix + 1,
+            left_arr_ix,
+            right_arr_ix + 1,
+        )
+    end
+end
+
+func bubble_sort_elements{range_check_ptr}(
+    entries_len : felt,
+    entries : TickElem*,
+    idx1 : felt,
+    idx2 : felt,
+    sorted_entries : TickElem*,
+    sorted_this_iteration : felt,
+) -> (sorted_entries : TickElem*):
+    alloc_locals
+    local entries : TickElem* = entries
+    local range_check_ptr = range_check_ptr
+
+    if idx2 == entries_len:
+        assert [sorted_entries + (idx2 - 1) * TickElem.SIZE] = [entries + idx1 * TickElem.SIZE]
+        if sorted_this_iteration == 0:
+            return (sorted_entries)
+        end
+
+        let (new_sorted_ptr : TickElem*) = alloc()
+        let (recursive_sorted_ptr) = bubble_sort_elements(
+            entries_len, sorted_entries, 0, 1, new_sorted_ptr, 0
+        )
+        return (recursive_sorted_ptr)
+    end
+    let (is_ordered) = is_le(
+        [entries + idx1 * TickElem.SIZE].tick, [entries + idx2 * TickElem.SIZE].tick
+    )
+    if is_ordered == TRUE:
+        assert [sorted_entries + (idx2 - 1) * TickElem.SIZE] = [entries + idx1 * TickElem.SIZE]
+        let (recursive_sorted_ptr) = bubble_sort_elements(
+            entries_len, entries, idx2, idx2 + 1, sorted_entries, sorted_this_iteration
+        )
+        return (recursive_sorted_ptr)
+    end
+    assert [sorted_entries + (idx2 - 1) * TickElem.SIZE] = [entries + idx2 * TickElem.SIZE]
+    let (recursive_sorted_ptr) = bubble_sort_elements(
+        entries_len, entries, idx1, idx2 + 1, sorted_entries, 1
+    )
+    return (recursive_sorted_ptr)
```

Comment:
Think we can delete as I cannot come up with a case where we would use bubble over merge

Reply:
it's a general sorting algorithm, just because we don't have a use case doesn't mean no one ever will.  It may also serve as a reference implementation.